code/mob/living/silicon/ai.dm

```diff
@@ -1886,6 +1886,38 @@ var/list/ai_emotions = list("Happy" = "ai_happy",\
 	else
 		boutput(usr, "<span class='alert'><b>Radio not found!</b></span>")
 
+/mob/living/silicon/ai/verb/open_map()
+	set name = "Open station map"
+	set desc = "Click on the map to teleport"
+	set category = "AI Commands"
+
+	var/datum/tgui/map_ui
```

Comment:
This is likely to do with why the UI is opening twice, since you're recreating the tgui datum every time the verb is called where in the original it's stored on the mob. Just moving this to be a mainframe var should fix it.

Reply:
This did the trick, thank you!